app/models/mentor_management.rb

```diff
@@ -0,0 +1,84 @@
+class MentorManagement
+  # Select a mentor using the following algorithm
+  #
+  # 1) Find all assignment participants for the
+  #    assignment with id [assignment_id] whose
+  #    duty is the same as [Particpant#DUTY_MENTOR].
+  # 2) Count the number of teams those participants
+  #    are a part of, acting as a proxy for the
+  #    number of teams they mentor.
+  # 3) Return the mentor with the fewest number of
+  #    teams they're currently mentoring.
+  #
+  # This method's runtime is O(n lg n) due to the call to
+  # Hash#sort_by. This assertion assumes that the
+  # database management system is capable of fetching the
+  # required rows at least as quickly.
+  #
+  # Implementation detail: Any tie between the top 2
+  # mentors is decided by the Hash#sort_by algorithm.
+  #
+  # @return The id of the mentor with the fewest teams
+  #   they are assigned to. Returns `nil` if there are
+  #   no participants with mentor duty for [assignment_id].
+  def self.select_mentor(assignment_id)
+    mentor_user_id, _ = self.zip_mentors_with_team_count(assignment_id).first
+    User.where(id: mentor_user_id).first
+  end
+
+  def self.update_mentor_state(assignment_id, team_id)
```

Comment:
considering rename to `assign_mentor()`?? add comments of what this method does.